### DISCUSSION TITLE
Nuget.exe and the dotnet tool

### DISCUSSION THREAD

**Discussion by overeemm** [cake-build/cake#2947] [Q&A] [answered]
I've started to migrate one of our repositories to cake 1.0.0 and the dotnet tool. Because we use dev.azure.com we need credentials on nuget feeds. Therefore I used the option `--nuget_useinprocessclient=false`. This options still seems to work, which is great. However

1. Can I put this configuration somewhere as a default? Otherwise all developers need to remember to add this option.
2. The `build.ps1` downloaded the nuget.exe if it could not be resolved. However, the dotnet tool does not do this. This makes it necessary to put nuget.exe on the path for every developer.

What do you recommend in this scenario?

**Comment by devlead** [comment, marked as answer]
Configuration ca be set through environment variables too, so they also be set on user/machine level
https://cakebuild.net/docs/running-builds/configuration/default-configuration-values

**Reply by devlead** [reply]
You could try tool directive to fetch NuGet
https://github.com/cake-build/cake/blob/b75c574e9acea3731eaf8ebe2d34b5b024efb1c9/build.cake#L13

**Reply by overeemm** [reply]
This is not a solution, because the first download already needs nuget.exe...

**Reply by devlead** [reply]
You're correct.

**Comment by overeemm** [comment]
Yes, the environment variable is a great solution!
